specs/CoreWebView2.CreateFromCOMObject.md

```diff
@@ -0,0 +1,82 @@
+CoreWebView2.CreateFromComObject
+===
+
+# Background
+The new [Unity WebView2](https://learn.microsoft.com/en-us/windows/mixed-reality/develop/
+advanced-concepts/webview2-unity-plugin) control creates and uses C++ COM to create and manage the
+ICoreWebView2* objects. However, Unity developers are often interacting with the Unity WebView2
+control using C#/.NET. The Unity WebView2 control doesn't expose the CoreWebView2 directly to devs
+using the Unity WebView2 control, so when devs want to call an API on CoreWebView2, they
+have to rely on that API being exposed on the Unity WebView2 control, which then internally calls 
+into CoreWebView2. This is in contrast to our other controls (like WPF WebView2 and Winforms
+WebView2 controls) which directly give access to their [CoreWebView2 object]
+(https://learn.microsoft.com/en-us/dotnet/api/microsoft.web.webview2.winforms.webview2.
+corewebview2?view=webview2-dotnet-1.0.2088.41), allowing devs to call any API that exists or gets
+added to the CoreWebView2/ICoreWebView2_*. The Unity WebView2 control can't do this today,
+as they are unable to create a CoreWebView2 object that wraps an already existing COM object.
+To help implement this for Unity, we are adding a new static factory function on CoreWebView2 .NET
+class that will allow it to wrap an existing ICoreWebView2 COM object, instead of creating a new
+one underlying ICoreWebView2.
+
+# Examples
+## CoreWebView2.CreateFromComObject
+```c#
+public class MyWebView2Control
+{
+  ... // Regular control code
+
+  CoreWebView2 _myCoreWebView2 = null;
+
+  [DllImport(DLL_NAME, CallingConvention = CallingConvention.StdCall)]
+  public static extern ComNativePointer GetNativePointer(WebViewInstancePtr instanceId);
+
+  // This is the CoreWebView2 property which allows developers to access CoreWebView2 APIs directly.
+  public CoreWebView2 CoreWebView2
+  {
+    get
+    {
+      if (!_myCoreWebView2)
+      {
+        IntPtr comPtr = WebViewNative.GetNativePointer(InstanceId);
```

Comment:
is InstanceId presumed to be a global variable?  Will most developers have something like this of type WebViewInstancePtr on hand?

Reply:
Developers using WebView2 with C++ (like Hololens Unity control) will have a pointer to their ICoreWebView2, or can get a pointer through their pointer to ICoreWebView2Controller. 

This is just sample code I got from the Unity team. They have some concept of an instance ID to find a particular WebView2 on their side, and can use it to get the ICoreWebView2 pointer associated with that WebView2. I think in their case the instance ID is global. It's not a public or common type, and we'd expect individual devs to have some function to get their ICoreWebView2 pointer from the native side.

If it helps with clarity I can omit the function/details to get the ICoreWebView2 pointer from native, and instead assume this code already has an IntPtr.

Reply:
I don't know .NET development well enough to know: if you're in the situation where this API is useful and our docs say to pass in a COM ICoreWebView2 pointer as an IntPtr, is that good enough for the docs? Its just the value of the COM interface pointer as an IntPtr and there's not additional .NET method calls to get that to happen?

Reply:
It looks like the sample leaks the COM pointer. The corresponding `Marshal.GetIUnknownForObject` method returns an AddRef'd COM pointer (in the form of an IntPtr) and you are expected to call `Marshal.Release()` on it when you're done.

Or does Unity's GetNativePointer behave differently? There are no docs for it, so I can't tell.